lib/full-site-editing/class-wp-rest-templates-controller.php

```diff
@@ -0,0 +1,426 @@
+				'href' => rest_url( $base ),
+			),
+			'about'      => array(
+				'href' => rest_url( 'wp/v2/types/' . $this->post_type ),
+			),
+		);
+
+		return $links;
+	}
+
+	/**
+	 * Get the link relations available for the post and current user.
+	 *
+	 * @return array List of link relations.
+	 */
+	protected function get_available_actions() {
+		$rels = array();
+
+		$post_type = get_post_type_object( $this->post_type );
+
+		if ( current_user_can( $post_type->cap->publish_posts ) ) {
+			$rels[] = 'https://api.w.org/action-publish';
+		}
+
+		if ( current_user_can( 'unfiltered_html' ) ) {
+			$rels[] = 'https://api.w.org/action-unfiltered-html';
+		}
+
+		return $rels;
+	}
+
+	/**
+	 * Retrieves the query params for the posts collection.
+	 *
+	 * @return array Collection parameters.
+	 */
+	public function get_collection_params() {
+		$query_params = parent::get_collection_params();
+
+		$query_params['theme'] = array(
+			'description' => __( 'Limit to the current theme templates.', 'gutenberg' ),
+			'type'        => 'string',
+		);
+		$query_params['wp_id'] = array(
+			'description' => __( 'Limit to the specified post id.', 'gutenberg' ),
+			'type'        => 'integer',
+		);
+
+		return $query_params;
+	}
+
+	/**
+	 * Retrieves the block type' schema, conforming to JSON Schema.
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => $this->post_type,
+			'type'       => 'object',
+			'properties' => array(
+				'id'          => array(
+					'description' => __( 'ID of template.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'slug'        => array(
+					'description' => __( 'Unique slug identifying the template.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'required'    => true,
+					'minLength'   => 1,
+					'pattern'     => '[a-zA-Z_\-]+',
+				),
+				'theme'       => array(
+					'description' => __( 'Theme identifier for the template.', 'gutenberg' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+				),
+				'is_custom'   => array(
+					'description' => __( 'Whether the template is customized.', 'gutenberg' ),
+					'type'        => 'bool',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'content'     => array(
+					'description' => __( 'Content of template.', 'gutenberg' ),
+					'type'        => 'string',
+					'default'     => '',
+					'context'     => array( 'embed', 'view', 'edit' ),
+				),
+				'title'       => array(
+					'description' => __( 'Title of template.', 'gutenberg' ),
+					'type'        => 'string',
```

Comment:
This should be an `object`. See the posts controller for reference or https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/#objects.

Reply:
If I set it to "object" updating doesn't work since it's passed a string. how does the post controller separate between the update and the get request for these things?

Reply:
The posts controller removes validation, but that isn't necessary anymore. You can instead do `type => array( 'object', 'string' )`